zerver/lib/upload.py

```diff
@@ -590,40 +587,28 @@ def get_realm_logo_url(self, realm_id: int, version: int, night: bool) -> str:
             file_name = "night_logo.png"
         return f"{self.avatar_bucket_url}/{realm_id}/realm/{file_name}?version={version}"
 
-    def ensure_medium_avatar_image(self, user_profile: UserProfile) -> None:
+    def ensure_avatar_image(
+        self, user_profile: UserProfile, is_medium: bool = False
+    ) -> None:  # nocoverage
+        file_extension = "-medium.png" if is_medium else ""
```

Comment:
In c03615b982734020d39fabbca63ab91781d5c485, the ".png" extension was removed for basic size avatars. Should I add that here?

Reply:
I'm not sure I understand the question.

I guess one thing to mention is that any changes to these functions requires migration code to move files to where they are expected to be -- so a refactor like this should aim to make no functional changes.

Reply:
So I think that this means we should preserve the `TODO` comment that we should transition this to `.png`, but not do said transition in this PR.

Reply:
Added the TODO comment.

Reply:
Related issue https://github.com/zulip/zulip/issues/12852